Banners GA (#9656)
braze-inc/braze-docs: _docs/_user_guide/message_building_by_channel/banners/testing.md

---
nav_title: Testing
article_title: Testing Banners
page_order: 3
description: "This reference article covers how to preview and test Banners, as well as some best practices."
channel:
  - banners
  
---

# Banner testing

> > It’s important to send a test Banner message before launching your campaign so you can ensure all media, copy, personalization, and custom attributes are generated correctly.

Reviewer: ```suggestion
> It’s important to send a test Banner message before launching your campaign so you can ensure all media, copy, personalization, and custom attributes render correctly.
```

Our preview and testing capabilities offer two ways to look at your Banners:

- You can preview your Banner as you compose it. This can help you visualize what your final message will look like from your user’s perspective. 
- Send a test message to yourself or a specific user’s device.

## Previewing your Banner

In the **Preview** tab of your composer, the view of your message might not be identical to its actual rendering on the user’s device.

![Preview tab of the Banner composer.]({% image_buster /assets/img/banners/preview_banner.png %})

## Sending a test message

When sending a test Banner, be sure to check for the following:

- The campaign must be assigned to a placement.

> - The placement must be requested from the user.

Reviewer 2: when you get the chance could either of you clarify what this means specifically? how does a user request this?

@niccrowley @mark-at-braze

Reviewer 3: The first one can probably be removed. A marketer can't get to the test tab--open the editor--without first picking a placement. @niccrowley can you confirm there aren't any edge cases here?

For the second, the integrator needs to have set up the integration to request the placement the marketer wants to test. I could create a placement, make a campaign assigned to that placement, and send a test message...but if the engineers haven't done the work to integrate that placement, it won't work.

Reviewer 2: @mark-at-braze  haha, i already removed the first one in my latest edit 🧠 

and for second one: it's just saying that placements need to be created per:

https://www.braze.com/docs/developer_guide/banners/creating_placements

correct?

Reviewer: It's not just creating the placement, but customer need to request the placement via the SDK in order to see the change

Reviewer 2: ah sorry, i meant: if they completed the steps on SDK side, then that would mean that this has already been done? @niccrowley

Reviewer: That's correct though they will need to ensure the placement their testing against is set in the SDK

- You can view Banners sent using the test send for up to 5 minutes on the assigned placement.

## Test checklist

- Is your banner campaign assigned to a placement?

> - Do the images and media show up and act as expected?

Reviewer 3: ```suggestion
- Do the images and media show up and act as expected on your targeted device types and screen sizes?
```

- Do your links and buttons direct the user to where they should go?
- Does the Liquid function as expected? Have you accounted for a default attribute value in the event that the Liquid returns no information?
- Is your copy clear, concise, and correct?

> 

Reviewer 3: @niccrowley 

A test message (the most recent test message) takes precedence for a placement, correct? Even if there is a high priority message the test user is eligible for, a test message will be shown first?

Reviewer: That's right

> - Do the images and media show up and act as expected?
> - Do your links and buttons direct the user to where they should go?
> - Does the Liquid function as expected? Have you accounted for a default attribute value in the event that the Liquid returns no information?
> - Is your copy clear, concise, and correct?
> - Is your banner campaign assigned to a placement?

Reviewer: ```suggestion
- Is your banner campaign assigned to a placement?
- Do the images and media show up and act as expected?
- Do your links and buttons direct the user to where they should go?
- Does the Liquid function as expected? Have you accounted for a default attribute value in the event that the Liquid returns no information?
- Is your copy clear, concise, and correct?
```

ensuring placementId should be first
